ports/rp2/machine_pin.c

```diff
@@ -261,155 +223,101 @@ static const mp_arg_t allowed_args[] = {
+        } else if (mode == MACHINE_PIN_MODE_OUT) {
             mp_hal_pin_output(self->id);
-        } else if (mode == GPIO_MODE_OPEN_DRAIN) {
+        } else if (mode == MACHINE_PIN_MODE_OPEN_DRAIN) {
             mp_hal_pin_open_drain(self->id);
         } else {
-            // Alternate function.
-            gpio_set_function(self->id, args[ARG_alt].u_int);
+            // Configure alternate function.
+            mp_uint_t af = args[ARG_alt].u_int;
+            if (machine_pin_find_af(self, af) == NULL) {
+                mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("Invalid pin af: %d."), af);
+            }
+            gpio_set_function(self->id, af);
             machine_pin_open_drain_mask &= ~(1 << self->id);
         }
     }
 
-    // configure pull (unconditionally because None means no-pull)
-    uint32_t pull = 0;
-    if (args[ARG_pull].u_obj != mp_const_none) {
-        pull = mp_obj_get_int(args[ARG_pull].u_obj);
+    if (!is_ext_pin(self)) {
+        // Configure pull (unconditionally because None means no-pull).
+        uint32_t pull = 0;
+        if (args[ARG_pull].u_obj != mp_const_none) {
+            pull = mp_obj_get_int(args[ARG_pull].u_obj);
+        }
+        gpio_set_pulls(self->id, pull & GPIO_PULL_UP, pull & GPIO_PULL_DOWN);
     }
-    gpio_set_pulls(self->id, pull & GPIO_PULL_UP, pull & GPIO_PULL_DOWN);
     return mp_const_none;
 }
 
 // constructor(id, ...)
 mp_obj_t mp_pin_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args) {
     mp_arg_check_num(n_args, n_kw, 1, MP_OBJ_FUN_ARGS_MAX, true);
-
-
     const machine_pin_obj_t *self = NULL;
+
     if (mp_obj_is_str(args[0])) {
         const char *name = mp_obj_str_get_str(args[0]);
-        #if MICROPY_HW_PIN_ENABLE_LED_PIN
-        if (!strcmp(name, LED_PIN_NAME)) {
-            self = led_pin_obj;
-        }
-        #endif
-        #if MICROPY_HW_PIN_CYW43_COUNT
-        static_assert(MICROPY_HW_PIN_CYW43_COUNT < 10, ""); // makes parsing name easy!
-        if (!self && !strncmp(name, CYW43_PIN_NAME_PREFIX, strlen(CYW43_PIN_NAME_PREFIX)) && strlen(name) == strlen(CYW43_PIN_NAME_PREFIX) + 1) {
-            int num = name[strlen(CYW43_PIN_NAME_PREFIX)] - '0';
-            if (num < MICROPY_HW_PIN_CYW43_COUNT) {
-                self = &cyw43_pin_obj[num];
-            }
+        // Try to find the pin in the board pins first.
+        self = machine_pin_find_named(&pin_board_pins_locals_dict, args[0]);
+        if (!self) {
+            // If not found, try to find the pin in the cpu pins which include
+            // CPU and and externally controlled pins (if any).
+            self = machine_pin_find_named(&pin_cpu_pins_locals_dict, args[0]);
         }
-        #endif
         if (!self) {
-            mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("Unknown named pin \"%s\""), name);
+            mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("Unknown named pin \"%s\"."), name);
```

Comment:
lowercase, no full-stop

Reply:
FYI existing error messages start with uppercase: https://github.com/micropython/micropython/blob/master/ports/rp2/machine_pin.c#L273

Not sure if you want me to fix them all or just the ones I added.

Reply:
Fixed all exception messages.